In [29]:
import pandas as pd
import numpy as np

In [30]:
# สร้าง dataframe ชื่อ data ว่างๆ ขึ้นมาก่อน
data = pd.DataFrame()

In [31]:
# สร้างข้อมูลที่เป็น class เก็บใน dataframe ที่สร้างไว้ 
data['play'] = ['No','No','Yes','Yes','Yes','No','Yes','No','Yes','Yes','Yes','Yes','Yes','No']

In [32]:
# สร้้างข้อมูลที่เป็น attribute
data['weather'] = ['Sunny','Sunny','Overcast','Rainy','Rainy','Rainy','Overcast','Sunny','Sunny','Rainy','Sunny','Overcast','Overcast','Rainy']
data['temp'] = ['Hot','Hot','Hot','Mild','Cool','Cool','Cool','Mild','Cool','Mild','Mild','Mild','Hot','Mild']
data['windy'] = ['Strong','Strong','Weak','Weak','Strong','Weak','Strong','Weak','Weak','Strong','Weak','Weak','Strong','Weak']

In [33]:
data

play   weather  temp   windy
0    No     Sunny   Hot  Strong
1    No     Sunny   Hot  Strong
2   Yes  Overcast   Hot    Weak
3   Yes     Rainy  Mild    Weak
4   Yes     Rainy  Cool  Strong
5    No     Rainy  Cool    Weak
6   Yes  Overcast  Cool  Strong
7    No     Sunny  Mild    Weak
8   Yes     Sunny  Cool    Weak
9   Yes     Rainy  Mild  Strong
10  Yes     Sunny  Mild    Weak
11  Yes  Overcast  Mild    Weak
12  Yes  Overcast   Hot  Strong
13   No     Rainy  Mild    Weak

In [34]:
# สร้าง dataframe ชื่อ person ว่างๆ ขึ้นมาก่อน
person = pd.DataFrame()

# สร้างข้อมูลที่เป็น new data 
person['weather'] = ['Rainy']
person['temp'] = ['Cool']
person['windy'] = ['Weak']

# View the data 
person

weather  temp windy
0   Rainy  Cool  Weak

In [35]:
# นับจำนวน yes
n_yes = len(data[data['play'] == 'Yes'])  #len(object) เป็น function ใช้นับจำนวน object

# นับจำนวน no
n_no = len(data[data['play'] == 'No']) 

# นับจำนวนที่เล่นทั้งหมด
total_play = len(data['play'])

print(n_yes,n_no,total_play)

9 5 14


In [36]:
n_sunny_yes = len(data[(data['weather'] == 'Sunny') & (data['play'] == 'Yes')])
n_rain_yes = len(data[(data['weather'] == 'Rainy') & (data['play'] == 'Yes')])
n_overcast_yes = len(data[(data['weather'] == 'Overcast') & (data['play'] == 'Yes')])

n_hot_yes = len(data[(data['temp'] == 'Hot') & (data['play'] == 'Yes')])
n_mild_yes = len(data[(data['temp'] == 'Mild') & (data['play'] == 'Yes')])
n_cool_yes = len(data[(data['temp'] == 'Cool') & (data['play'] == 'Yes')])

n_weak_yes = len(data[(data['windy'] == 'Weak') & (data['play'] == 'Yes')])
n_strong_yes = len(data[(data['windy'] == 'Strong') & (data['play'] == 'Yes')])

In [37]:
n_sunny_no = len(data[(data['weather'] == 'Sunny') & (data['play'] == 'No')])
n_rain_no = len(data[(data['weather'] == 'Rainy') & (data['play'] == 'No')])
n_overcast_no = len(data[(data['weather'] == 'Overcast') & (data['play'] == 'No')])

n_hot_no= len(data[(data['temp'] == 'Hot') & (data['play'] == 'No')])
n_mild_no = len(data[(data['temp'] == 'Mild') & (data['play'] == 'No')])
n_cool_no = len(data[(data['temp'] == 'Cool') & (data['play'] == 'No')])

n_weak_no = len(data[(data['windy'] == 'Weak') & (data['play'] == 'No')])
n_strong_no = len(data[(data['windy'] == 'Strong') & (data['play'] == 'No')])

In [38]:
p_yes = n_yes/total_play 
p_no = n_no/total_play 

p_sunny_yes = n_sunny_yes/n_yes
p_rain_yes = n_rain_yes/n_yes
p_overcast_yes = n_overcast_yes/n_yes

p_hot_yes = n_hot_yes/n_yes
p_mild_yes = n_mild_yes/n_yes
p_cool_yes = n_cool_yes/n_yes

p_weak_yes = n_weak_yes/n_yes
p_strong_yes = n_strong_yes/n_yes

print(p_rain_yes)

0.3333333333333333


In [39]:
p_sunny_no = n_sunny_no/n_no
p_rain_no = n_rain_no/n_no
p_overcast_no = n_overcast_no/n_no

p_hot_no = n_hot_no/n_no
p_mild_no = n_mild_no/n_no
p_cool_no = n_cool_no/n_no

p_weak_no = n_weak_no/n_no
p_strong_no = n_strong_no/n_no

print(p_rain_no)

0.4


In [40]:
p_rain_cool_weak_yes = p_rain_yes * p_cool_yes * p_weak_yes * p_yes

p_rain_cool_weak_no = p_rain_no * p_cool_no * p_weak_no * p_no

print(p_rain_yes,p_cool_yes,p_weak_yes,p_yes)
print(p_rain_no,p_cool_no,p_weak_no,p_no)

print('p_rain_cool_weak_yes = ', p_rain_cool_weak_yes)
print('p_rain_cool_weak_no = ', p_rain_cool_weak_no)

0.3333333333333333 0.3333333333333333 0.5555555555555556 0.6428571428571429
0.4 0.2 0.6 0.35714285714285715
p_rain_cool_weak_yes =  0.03968253968253968
p_rain_cool_weak_no =  0.017142857142857147


ค่า P(play = no|new) (=0.0171) มีค่าน้อยกว่า P(play = yes|new) (=0.0397) ดังนั้นโมเดลจึงทำนายว่าข้อมูลตัวอย่างนี้มีค่าคลาส play = yes เล่น tennis